src/components/SavedFilters.vue

```diff
@@ -39,17 +39,26 @@
     return allOptions
   })
 
+  const filterInRoute = computed<SavedSearchFilter>(() => ({
+    state: filter.flowRuns.state.name ?? [],
+    flow: filter.flows.id ?? [],
+    deployment: filter.deployments.id ?? [],
+    workPool: filter.workPools.name ?? [],
+    tag: filter.flowRuns.tags.name ?? [],
+    startDate: filter.flowRuns.expectedStartTimeAfter != undefined ? String(filter.flowRuns.expectedStartTimeAfter) : undefined,
+    endDate: filter.flowRuns.expectedStartTimeBefore != undefined ? String(filter.flowRuns.expectedStartTimeBefore) : undefined,
+  }))
+
+  const { value: myCustomDefaultFilter } = useCustomDefaultFlowRunsFilter()
+  watch(filterInRoute, (newValue) => {
+    if (myCustomDefaultFilter.value !== null && isEmptyFilter(newValue)) {
+      setFilters(myCustomDefaultFilter.value)
```

Comment:
This `setFilters` does a `router.push` under the hood. We probably want to find a way to do a `router.replace` for this so we don't end up with double history

Reply:
👀 
ah yeah that feels like having a custom default as-is would probably get you into a bit of a loop trying to click back

Reply:
ah I did try adding a beforeEnter/nav guard to basically do that. that... worked but this seemed maybe a little better. 

are you thinking like [here](https://github.com/PrefectHQ/prefect-ui-library/blob/0fa90a173b1abdcde6722cdc6097e1b105e2f019/src/router/routes.ts#L13-L14)?

Reply:
Correct, so anywhere that method is called it returns a url with the filters already. I THINK since you're storing the filters themselves in local storage you could grab and apply them just fine. Just an idea, I don't have much opinion here.

Reply:
Hmmm iirc I also went away from the nav guard b/c the guard didn't rerun on query-param-only route changes so clicking on Flow Runs from the nav menu after already having filters wouldn't trigger my custom default. I think this suggestion would fix that but on the other side navigating directly to /flow_runs somehow would probably circumvent a check here? 

oh also I was struggling to figure out how to go from `const filter: FlowRunsFilter` -> `router.push({..., query: dotFlatten(filter))` w/o the `useRouteQueryParams` magic that's under the hood of the `setFilters` here

Reply:
Hmm right, yeah that wouldn't be as simple as I'm I was thinking because there isn't a utility for converting the filters to just query params. 

Might be a nice enhancement to that stuff. I'll look into that if you think it would be helpful here

Reply:
Thinking of kicking the can on this for now - not ideal but breaking back just for user-defined custom view for _now_ seems _okayyyy_. It'd be _neat_ if `setFilters` allowed me to just overwrite (which would under-the-hood replace) and looked into the route query compositions and seems non-trivial. 

alternatively hooking into router configurations here feels less correct to me than having the `<SavedFilter />` own the responsibility of having/setting a custom, default saved filter. 

may address in a follow-up PR.

Reply:
resolved in [8dfa717](https://github.com/PrefectHQ/prefect-ui-library/pull/1753/commits/8dfa717157135268ce8ae21f056db6e94cfdf1af)